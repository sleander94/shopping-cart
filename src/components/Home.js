import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Home.css';

const Home = () => {
  return (
    <div id="home">
      <div className="homeTitle1">
        Our favorite places are full of surprises...
      </div>
      <div className="homeTitle2">The right gear means everything.</div>
      <div className="homeTitle3">Be prepared for any challenge.</div>
      <div className="imageContainer">
        <img
          src={require('../images/todd-diemer-fpNBYsymggk-unsplash.jpg')}
        ></img>
        <div className="credit">Photo by Todd Diemer on Unsplash</div>
      </div>
      <Link to="/shop">Shop now</Link>
    </div>
  );
};

export default Home;
