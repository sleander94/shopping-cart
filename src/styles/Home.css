#home {
  display: grid;
  grid: 1fr 1fr 1fr 1fr / 1fr 1fr;
  justify-items: center;
  align-items: center;
  column-gap: 24px;
  padding: 72px;
  height: 100%;
  text-align: center;
  background: rgb(240, 240, 240);
}

#home .imageContainer {
  grid-area: 1 / 2 / 5 / 3;
}

#home img {
  width: 700px;
  height: auto;
  border-radius: 4px;
  border: 1px solid gray;
}

#home .credit {
  position: relative;
  left: 470px;
  bottom: 25px;
  width: 240px;
  font-size: 0.8rem;
}

.homeTitle1 {
  grid-area: 1 / 1 / 2 / 2;
  font-size: 2.5rem;
  color: rgb(94, 92, 92);
}
.homeTitle2 {
  grid-area: 2 / 1 / 3 / 2;
  font-size: 3.5rem;
}

.homeTitle3 {
  grid-area: 3 / 1 / 4 / 2;
  font-size: 2rem;
  font-weight: 700;
}

#home a {
  grid-area: 4 / 1 / 5 / 2;
  background: black;
  color: white;
  border: 1px solid gray;
  text-decoration: none;
  font-size: 1.8rem;
  padding: 14px 48px;
  border-radius: 12px;
}

#home a:hover {
  background: rgba(0, 0, 0, 0.714);
}
