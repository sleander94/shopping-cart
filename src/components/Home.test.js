import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import Home from './Home';

describe('Home component', () => {
  it('renders correctly', () => {
    const home = renderer.create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
    expect(home).toMatchSnapshot();
  });

  it('links to store', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
    userEvent.click(screen.getByRole('link'));
    expect(window.location.href).toContain('/shop');
  });
});
