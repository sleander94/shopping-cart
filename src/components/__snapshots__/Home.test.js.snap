// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  id="home"
>
  <div
    className="homeTitle1"
  >
    Our favorite places are full of surprises...
  </div>
  <div
    className="homeTitle2"
  >
    The right gear means everything.
  </div>
  <div
    className="homeTitle3"
  >
    Be prepared for any challenge.
  </div>
  <div
    className="imageContainer"
  >
    <img
      src="todd-diemer-fpNBYsymggk-unsplash.jpg"
    />
    <div
      className="credit"
    >
      Photo by Todd Diemer on Unsplash
    </div>
  </div>
  <a
    href="/shop"
    onClick={[Function]}
  >
    Shop now
  </a>
</div>
`;
